### ISSUE TITLE
Cannot use GcsService, because of old guava dependency.

### ISSUE THREAD

**Issue by fipil** [GoogleCloudPlatform/appengine-gcs-client#23]
I ran to this problem, see the error:
java.lang.NoSuchMethodError: com.google.common.base.Stopwatch.createUnstarted()Lcom/google/common/base/Stopwatch;
	at com.google.appengine.tools.cloudstorage.RetryHelper.runWithRetries(RetryHelper.java:156)
	at com.google.appengine.tools.cloudstorage.GcsServiceImpl.createOrReplace(GcsServiceImpl.java:70)

I was searching a lot what is the problem and I think that it's guava dependency problem.
See this issue: https://github.com/google/google-api-java-client/issues/903

IMHO appengine-gcs-client uses the google-api-java-client, which used guava v13.
The guava v13 doesn't contain the Stopwatch.createUnstarted() method, it's available from v15.

As written in the issue 903 (link is above), author updated google-api-java-client to use the guava v15.

It's a big problem, because I cannot use GcsService anymore.
Thanks.
Filip.

**Comment by aozarov** [comment]
The `appengine-gcs-client` [POM](https://github.com/GoogleCloudPlatform/appengine-gcs-client/blob/master/java/pom.xml#L88) explicitly excludes `guava-jdk5` from being an indirect dependency.

Also, the upper bound of the `google-api-client-appengine` should include the most recent version.

Are you sure you don't pickup `guava-jdk5` via other dependencies?

**Comment by fipil** [comment]
![image](https://cloud.githubusercontent.com/assets/768590/11326032/31e7e07a-915e-11e5-83a2-c1a0277692fe.png)
Looking to dependency tree I see that the google-api-client-engine depends on the guava-jdk5.

**Comment by fipil** [comment]
I just tried to add the exclusion og guava-jdk5 to all google dependencies in my POM. I also put the guava dependency at the beginning:
&lt;dependency&gt;
   &lt;groupId&gt;com.google.guava&lt;/groupId&gt;
   &lt;artifactId&gt;guava&lt;/artifactId&gt;
   &lt;version&gt;[18.0]&lt;/version&gt;
&lt;/dependency&gt;
Then my dependency tree filtered to "guava" looks like this:
![image](https://cloud.githubusercontent.com/assets/768590/11326152/d777e078-9161-11e5-997c-3a4d3cbccf22.png)

Application works without problem running on the development server, but after publishing to appengine I again received this error:
![image](https://cloud.githubusercontent.com/assets/768590/11326163/1e172e1c-9162-11e5-8663-3ea38dca6b5c.png)

**Comment by fipil** [comment]
Ok, problem solved. I'm sorry for disturbing you. 
Problem was the following: 
somehow Eclipse didn't delete all jars from the target/MyAppWeb-1.1/WEB-INF/lib/ folder before compiling new WAR during deployment to appengine. So there was still file guava-jdk5.jar, no matter that I excluded it from dependencies. Some process was holding these jars because I had to restart windows to be able to delete them. After that and fresh recompiling and redeployment the error doesn't appear anymore.
Thank you for your time aozarov.
Filip.
